app/src/main/java/com/example/tasky/common/presentation/AgendaCard.kt

```diff
@@ -108,12 +120,31 @@ fun CustomCheckbox(isChecked: Boolean, color: Color, size: Dp) {
 }
 
 @Composable
-fun HorizontalEllipsisIcon(tint: Color, toggleAgendaCardDropdownVisibility: () -> Unit) {
-    IconButton(onClick = { toggleAgendaCardDropdownVisibility() }) {
-        Icon(
-            painter = painterResource(id = R.drawable.ic_more_horizontal),
-            contentDescription = stringResource(R.string.more_options),
-            tint = tint
+fun HorizontalEllipsisIcon(
+    tint: Color,
+    toggleAgendaCardDropdownVisibility: () -> Unit,
+    agendaItem: AgendaItem,
+    showAgendaCardDropdown: Boolean,
+    onAgendaCardActionClick: (AgendaItem, CardAction) -> Unit,
+    visibleAgendaCardDropdownId: String?,
+    setVisibleAgendaItemId: (String) -> Unit
+) {
+    Box {
+        IconButton(onClick = {
+            setVisibleAgendaItemId(agendaItem.id)
```

Comment:
Is this extra function really needed? After all, you only care about when the drop down was toggled. You do have access to `agendaItem.id` in the parent composable of this as well, so you don't need to explicitly pass it.